storehouse.py

```diff
@@ -20,15 +25,19 @@ def __init__(self, controller):
         """Create a storehouse client instance."""
         self.controller = controller
         self.namespace = 'kytos.topology.status'
-        self.list_stored_boxes()
+        self.wait_persistence_box = getattr(settings,
+                                            'WAIT_PERSISTENCE_BOX_TIMER',
+                                            DEFAULT_WAIT_PERSISTENCE_BOX_TIMER)
+
         if 'box' not in self.__dict__:
             self.box = None
+        self.list_stored_boxes()
 
     def get_data(self):
         """Return the persistence box data."""
-        if not self.box:
-            return {}
-        self.get_stored_box(self.box.box_id)
+        # Wait retrieve or create box in storehouse
+        while not self.box:
+            time.sleep(self.wait_persistence_box)
```

Comment:
The default here is 0.1 seconds, so a debug may flood the console. Maybe a better option is to add an upper limit: it runs for X times without result, give up and log and error.

Reply:
Resolved.